src/CKEditor.tsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {
+	eventNameToHandlerName,
+	defaultEvents,
+	stripPrefix,
+	handlerNameToEventName
+} from './events';
+import useCKEditor from './useCKEditor';
+import { camelToKebab } from './utils';
+
+import {
+	CKEditorEventDispatcher,
+	CKEditorEventHandlerProp,
+	CKEditorProps,
+	CKEditorType
+} from './types';
+
+const { useEffect, useRef, useState } = React;
+
+/**
+ * `CKEditor` component is a convenient wrapper around low-level hooks.
+ * It's useful for simpler use cases. For advanced usage see `useCKEditor` hook.
+ */
+function CKEditor<EventHandlerProp>( {
+	config,
+	debug,
+	editorUrl,
+	initData,
+	name,
+	readOnly,
+	style,
+	type,
+
+	/**
+	 * `handlers` object must contain event handlers props only!
+	 */
+	...handlers
+}: CKEditorProps<EventHandlerProp> ): JSX.Element {
+	/**
+	 * Uses `useState` instead of `useRef` to force re-render.
+	 */
+	const [ element, setElement ] = useState<HTMLDivElement | null>( null );
+
+	/**
+	 * Ensures referential equality of event handlers.
+	 */
+	const refs = useRef( handlers );
+
+	const dispatchEvent: CKEditorEventDispatcher = ( { type, payload } ) => {
+		const handlerName = eventNameToHandlerName(
+			stripPrefix( type )
+		) as keyof CKEditorEventHandlerProp;
+		const handler = refs.current[ handlerName ];
+
+		if ( handler ) {
+			handler( payload );
+		}
+	};
+
+	/**
+	 * `readOnly` prop takes precedence over `config.readOnly`.
+	 */
+	if ( config && typeof readOnly === 'boolean' ) {
+		config.readOnly = readOnly;
+	}
+
+	const { editor, status } = useCKEditor( {
+		config,
+		dispatchEvent,
+		debug,
+		editorUrl,
+		element,
+
+		/**
+		 * Subscribe only to those events for which handler was supplied.
+		 */
+		subscribeTo: Object.keys( handlers )
+			.filter( key => key.indexOf( 'on' ) === 0 )
+			.map( handlerNameToEventName ),
+		type
+	} );
+
+	/**
+	 * Sets and updates styles.
+	 */
+	useEffect( () => {
+		const canSetStyles =
+			type !== 'inline' &&
+			editor &&
+			( status === 'loaded' || status === 'ready' );
+
+		if ( style && canSetStyles ) {
+			editor.container.setStyles( style );
+		}
+
+		return () => {
+			if ( style && canSetStyles ) {
+				Object.keys( style )
+					.map( camelToKebab )
+					.forEach( styleName => {
+						editor.container.removeStyle( styleName );
+					} );
+			}
+		};
+	}, [ editor, status, style, type ] );
```

Comment:
It seems that the component allows adding styles only to classic editors, but in tests and samples I can see that styles are also added to inline editors.

Reply:
Styling for inline editor [is set on root `div` element](https://github.com/ckeditor/ckeditor4-react/blob/48e916c5783a1edd94d30c0d04caf30d475f13ba/src/CKEditor.tsx#L123) via regular `style` prop. My understanding is that in inline that root `div` is used by the editor while in classic mode a new root for editor is created. Therefore, for classic mode we have to imperatively change styles.